tests/components/ista_ecotrend/test_config_flow.py

```diff
@@ -6,14 +6,18 @@
 import pytest
 
 from homeassistant.components.ista_ecotrend.const import DOMAIN
+from homeassistant.components.recorder import Recorder
 from homeassistant.config_entries import SOURCE_USER
 from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
 from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResultType
 
 
 async def test_form(
-    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_ista: MagicMock
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_ista: MagicMock,
```

Comment:
You mean that i added recorder_mock? The statistics import is triggered in async_added_to_hass and does an initial statistics import, that's why the test fails if the recorder isn't initialized first.

Reply:
Tests for config-flow should be patched to avoid calling `async_added_to_hass` altogether.

Look for `mock_setup_entry` in the code for a sample fixture (usually it is auto-use in the config-flow tests)

Reply:
Oh, i understand now. async_added_to_hass is in fact patched out but i found out the reason the test fails is because the recorder is defined as dependency in the manifest.  If I change it to after_dependencies the test runs without the recorder fixture but not quite sure if that makes any difference for the functionality of the integration.

Reply:
it's in the conftest.py 
- https://github.com/home-assistant/core/blob/edd3c45c09b34715e180070435db75463d31228a/tests/components/ista_ecotrend/conftest.py#L29-L34

Reply:
The fixture is declared - but it's never activated.
Your are missing `pytestmark = pytest.mark.usefixtures("mock_setup_entry")` in `test_config_flow.py`

Reply:
Sorry - cancel everything I said - I was looking in the wrong place!